registers/mapper/main.go

```diff
@@ -0,0 +1,208 @@
+package main
+
+import (
+	"context"
+	"crypto/sha256"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"log"
+	"reflect"
+	"time"
+
+	"github.com/golang/groupcache/lru"
+	"github.com/google/trillian"
+	"github.com/google/trillian-examples/registers/trillian_client"
+	"google.golang.org/grpc"
+)
+
+var (
+	trillianLog = flag.String("trillian_log", "localhost:8090", "address of the Trillian Log RPC server.")
+	logID       = flag.Int64("log_id", 0, "Trillian LogID to read.")
+	trillianMap = flag.String("trillian_map", "localhost:8095", "address of the Trillian Map RPC server.")
+	mapID       = flag.Int64("map_id", 0, "Trillian MapID to write.")
+)
+
+type record struct {
+	Entry map[string]interface{}
+	Items []map[string]interface{}
+}
+
+// add only adds if the item is not already present in Items.
+func (r *record) add(i map[string]interface{}) {
+	for _, ii := range r.Items {
+		if reflect.DeepEqual(i, ii) {
+			return
+		}
+	}
+	r.Items = append(r.Items, i)
+}
+
+type recordCache struct {
+	records *lru.Cache
+	mapID   int64
+	tc      trillian.TrillianMapClient
+	ctx     context.Context
+}
+
+func newCache(tc trillian.TrillianMapClient, mapID int64, max int, ctx context.Context) *recordCache {
+	c := &recordCache{records: lru.New(max), mapID: mapID, tc: tc, ctx: ctx}
+	c.records.OnEvicted = func(key lru.Key, value interface{}) { recordEvicted(c, key, value) }
+	return c
+}
+
+func recordEvicted(c *recordCache, key lru.Key, value interface{}) {
+	fmt.Printf("evicting %v -> %v\n", key, value)
+
+	v, err := json.Marshal(value)
+	if err != nil {
+		log.Fatalf("Marshal() failed: %v", err)
+	}
+
+	hash := sha256.Sum256([]byte(key.(string)))
+	l := trillian.MapLeaf{
+		Index:     hash[:],
+		LeafValue: v,
+	}
+
+	req := trillian.SetMapLeavesRequest{
+		MapId:  c.mapID,
+		Leaves: []*trillian.MapLeaf{&l},
+	}
+
+	_, err = c.tc.SetLeaves(c.ctx, &req)
```

Comment:
Since you're discarding the response, you can make this `if _ err := c.tc.Set...; if err != nil { ... }`